ward/_diff.py

```diff
@@ -86,14 +86,17 @@ def rewrite_line(self, line, line_to_rewrite, prev_marker):
     def build_unified_diff(self) -> RenderResult:
         diff = self.raw_unified_diff()
         prev_marker = ""
-        output_lines = []
+        output_lines: List[Union[Text, str]] = []
         for line in diff:
             if line.startswith("- "):
                 output_lines.append(Text(line[2:], style="green"))
             elif line.startswith("+ "):
                 output_lines.append(Text(line[2:], style="red"))
             elif line.startswith("? "):
-                line_to_rewrite = output_lines[-1].plain
+                if isinstance(output_lines[-1], str):
+                    line_to_rewrite = output_lines[-1]
+                else:
+                    line_to_rewrite = output_lines[-1].plain
                 output_lines[-1] = self.rewrite_line(line, line_to_rewrite, prev_marker)
             else:
                 output_lines.append(line[2:])
```

Comment:
This I'm not 100% sure about. Maybe line 99 (before changes) should instead append a `Text` object instead of `str`?

Reply:
I think we should try to avoid `Union` where possible, so always appending `Text` (with no args other than the string) instead of `str` would make sense. They'll be treated the same by Rich anyway.

Reply:
That makes tests fail however, so treatment is not 100% the same :smile: 

Maybe this should be improved upon in a follow-up?

Reply:
I'll take a look at the tests and push a fix to this branch (assuming it's a simple fix, which I think it is)

Reply:
I've pushed up a change to always use `Text` and deleted the `if isinstance` block that was added. Let me know if there's any issues with that

Reply:
Looks great, this is also my preferred solution as long as tests pass!